packages/components/managed-data-table/src/managed-data-table.js

```diff
@@ -0,0 +1,148 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import requiredIf from 'react-required-if';
+import Spacings from '@commercetools-uikit/spacings';
+import DataTable from '@commercetools-uikit/data-table';
+import DataTableSettings from './data-table-settings';
+
+const ManagedDataTable = (props) => {
+  const columns = React.useMemo(
+    () =>
+      props.columns.map((column) => ({
+        ...column,
+        isTruncated: !props.displaySettings.disableDisplaySettings
+          ? props.displaySettings.isWrappingText
+          : column.isTruncated,
+      })),
+    [
+      props.columns,
+      props.displaySettings.disableDisplaySettings,
+      props.displaySettings.isWrappingText,
+    ]
+  );
+
+  return (
+    <Spacings.Stack>
+      <DataTableSettings
+        topBar={props.topBar}
+        onSettingsChange={props.onSettingsChange}
+        columnManager={props.columnManager}
+        displaySettings={props.displaySettings}
+      />
+      <DataTable
+        rows={props.rows}
+        columns={columns}
+        footer={props.footer}
+        onRowClick={props.onRowClick}
+        maxHeight={props.maxHeight}
+        maxWidth={props.maxWidth}
+        isCondensed={props.displaySettings.isCondensed}
+        disableHeaderStickiness={props.disableHeaderStickiness}
+        disableFooterStickiness={props.disableFooterStickiness}
+        itemRenderer={props.itemRenderer}
+        isHeaderSticky={props.isHeaderSticky}
+        wrapHeaderLabels={props.wrapHeaderLabels}
+        verticalCellAlignment={props.verticalCellAlignment}
+        horizontalCellAlignment={props.horizontalCellAlignment}
+        sortedBy={props.sortedBy}
+        onSortChange={props.onSortChange}
+        sortDirection={props.sortDirection}
+      />
```

Comment:
I've been thinking about this and I'm not sure this is the best approach.

I think it would make more sense to NOT include the `<DataTable>` component here but to allow the user to pass it as a child.
So the `<ManagedDataTable>` component becomes a compound component.

This has the advantage to NOT having to proxy all the props of the `<DataTable>` component and also to not be bound to that.
So for instance, users could choose another table if they want to, while still keeping the functionality of the state management.

The `columns` prop will be then injected into the children.

```js
<Spacings.Stack>
  <DataTableSettings />
  {React.cloneElement(props.children, { columns })}
</Spacings.Stack>
```

and you render the component as

```js
<ManagedDataTable>
  <DataTable />
</ManagedDataTable>
```

Reply:
I support this approach. It spares us from duplicating a lot of stuff and even allows using another Table component.

Note that for this we also need to inject the `isCondensed` prop.